x-pack/plugins/painless_lab/server/routes/api/execute.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { schema } from '@kbn/config-schema';
+
+import { RouteDependencies } from '../../types';
+import { API_BASE_PATH } from '../../../common/constants';
+import { isEsError } from '../../lib';
+
+const bodySchema = schema.object({
+  script: schema.object({
+    source: schema.string(),
+  }),
+});
+
+export function registerExecuteRoute({ router, license }: RouteDependencies) {
+  router.post(
+    {
+      path: `${API_BASE_PATH}/execute`,
+      validate: {
+        body: bodySchema,
+      },
+    },
+    license.guardApiRoute(async (ctx, req, res) => {
+      const body = req.body as typeof bodySchema.type;
```

Comment:
I've also run into this 😄. The higher order function (in this case `license.guardApiRoute`) needs to accept the three RequestHandler generic params `<P, Q, B>` and pass them through to its own handler and request object.

Like here:
https://github.com/elastic/kibana/blob/babf81bdc02174145f0416e579f8e18a8d27c2ed/x-pack/plugins/watcher/server/lib/license_pre_routing_factory/license_pre_routing_factory.ts#L15

We can also review this down the line.